gradle-metric-schema/src/main/java/com/palantir/metric/schema/gradle/DependencyRequirements.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.metric.schema.gradle;
+
+import com.google.common.collect.ImmutableSetMultimap;
+import com.google.common.collect.Multimap;
+import com.palantir.metric.schema.lang.LangMetricSchema;
+import com.palantir.metric.schema.lang.MetricSchemaCompiler;
+import java.io.File;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.Optional;
+
+public final class DependencyRequirements {
+
+    public static Multimap<String, String> getDependencies(Collection<File> metricSchemaFiles) {
+        if (metricSchemaFiles == null || metricSchemaFiles.isEmpty()) {
+            return ImmutableSetMultimap.of();
+        }
+
+        Collection<LangMetricSchema> rawSchemas = metricSchemaFiles.stream()
+                .<Optional<LangMetricSchema>>map(schemaFile -> {
+                    try {
+                        return Optional.of(MetricSchemaCompiler.parseRawSchema(schemaFile));
+                    } catch (IOException e) {
+                        // for configuration purposes, ignore files that cannot be parsed as schemas
+                        return Optional.empty();
+                    }
+                })
+                .<LangMetricSchema>mapMulti(Optional::ifPresent)
+                .toList();
+
+        if (rawSchemas.isEmpty()
+                || rawSchemas.stream().allMatch(schema -> schema.namespaces().isEmpty())) {
```

Comment:
`allMatch` on an empty collection is true.

```suggestion
        if (rawSchemas.stream().allMatch(schema -> schema.namespaces().isEmpty())) {
```